lib/drivers/insuletDriver.js

```diff
@@ -788,12 +788,19 @@ module.exports = function (config) {
       }
       // handle non-history alarms
       else {
-        // alarm.alarm.alarm is not a typo!
-        if (alarm.alarm.alarm_text != null) {
-          alarmValue = alarm.alarm.alarm;
+        // will occur for non-history alarms that aren't bootstrappable
+        // since we're no longer erroring on failure to look up UTC info
+        if (postalarm.time == '**REQUIRED**') {
+          postalarm = null;
         }
         else {
-          postalarm = null;
+          // alarm.alarm.alarm is not a typo!
+          if (alarm.alarm.alarm_text != null) {
+            alarmValue = alarm.alarm.alarm;
+          }
+          else {
+            postalarm = null;
+          }
```

Comment:
Uhm, would it maybe improve the readability here if we use
```
if {}
else if {}
else {}
```
instead of the nested ifs and elses?